### DISCUSSION TITLE
Frequency measuring component?

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#1192] [Ideas]
What do you think of the general idea of being able to measure frequencies?  

Imagine you're designing a VGA or PAL-TV interface and you want to check that the generated sync signals are correct. Then you could put a counter on the clock and single-step a gazillion times until the sync with toggles and then check how many clock ticks that took from there calculate the time or at least see that the count value is as expected.

With a small addition to the "counting-probe" it could freeze, display and reset the count value at regular intervals. The displayed value would then not be the raw count but rather calculated from the count, a value (set in the probe) that's the clock frequency IRL for the design, and the frequency of the Clock-component.

**Comment by hneemann** [comment]
It would require to connect a clock signal. 
Since there is no time in the simulator itself, since it works purely event-based, one could not otherwise make any statements about a frequency. The frequency would then be obtained as a fraction of the applied clock frequency.

**Reply by mengstr** [reply]
The clock signal would definitely be needed. As I said "calculated from the **count**, a value (set in the probe) that's the clock **frequency IRL** for the design, and the **frequency of the Clock** -component."

But in really wouldn't be needed to be hooked up to the Clock. Considering the limitations of the Clock-component already (only one max-speed, and you can only have one clock if there's a MonoFlop in the design) this frequency meter could be just as the MonoFlop - I.E using the "one and only" clock for the timebase.
